internal/server/handlers_certificate_authorities.go

```
@@ -478,3 +495,70 @@ func SignCertificateAuthority(env *HandlerConfig) http.HandlerFunc {
 		}
 	}
 }
+
+// GetCertificateAuthorityCRL handler returns the CRL of the associated CA
+// It returns a 200 OK on success
+func GetCertificateAuthorityCRL(env *HandlerConfig) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		id := r.PathValue("id")
+		idNum, err := strconv.ParseInt(id, 10, 64)
+		if err != nil {
+			writeError(w, http.StatusBadRequest, "Invalid ID")
+			return
+		}
+
+		ca, err := env.DB.GetDenormalizedCertificateAuthority(db.ByCertificateAuthorityID(idNum))
+		if err != nil {
+			log.Println(err)
+			if errors.Is(err, db.ErrNotFound) {
+				writeError(w, http.StatusNotFound, "Not Found")
+				return
+			}
+			writeError(w, http.StatusInternalServerError, "Internal Error")
+			return
+		}
+		crlResponse := CRL{CRL: ca.CRL}
+
+		err = writeResponse(w, crlResponse, http.StatusOK)
+		if err != nil {
+			writeError(w, http.StatusInternalServerError, "internal error")
+			return
+		}
+	}
+}
+
+// RevokeCertificateAuthorityCertificate handler receives an id as a path parameter,
+// and revokes the corresponding certificate by placing the certificate serial number to the CRL
+// It returns a 200 OK on success
+func RevokeCertificateAuthorityCertificate(env *HandlerConfig) http.HandlerFunc {
```

Notary won't be able to revoke a CA that was signed outside of it - we should probably return a specific error when someone tries to do this.

---

Added: "tcertificates need to be signed by a notary managed certificate authority in order to be revoked"